### DISCUSSION TITLE
use client and use server sintaxix

### DISCUSSION THREAD

**Discussion by dsaza** [vercel/next.js#57426] [Ideas]
### Goals

1. Improve syntaxix
2. Avoid errors

### Non-Goals

1. Possibilities from the core

### Background

I've noticed that many developers are dissatisfied with the syntax of 'use server' and 'use client.' They tend to forget to add this text at some point, which can lead to sensitive data leaking.

It would be a good idea to determine whether a component should be rendered on the client using a prop, for example:

```js
<Button client={true} />
<Button client="only" />
```

Regarding 'use server,' I would suggest requiring the creation of server code files in a specific folder, just like how 'app' and its router work. We could create a folder called 'server' so that any module imported from there is considered as 'use server.'

### Proposal

At the moment I would have no answer to this

**Comment by icyJoseph** [comment]
Hi,

IIRC from the RFC's where all of this was discussed, the key point is that the `module` is the network boundary. 

Far too often developers have mixed server/edge/client concerns without much thought and it shows with Server Components.

Also Client Components are not necessarily only rendered clients side, rather they are also SSR'd.

Again, the difference is that Server Components, do not expose their module to the client side bundle.

You can read through the arguments posed here: https://github.com/reactjs/rfcs/pull/227
